test/libsolidity/syntaxTests/parsing/address_invalid_state_mutability.sol

```diff
@@ -0,0 +1,34 @@
+contract C {
+	address view m_a;
+	address pure m_b;
+	address view[] m_c;
+	mapping(uint => address view) m_d;
+	function f() public pure {
+		address view a;
+		address pure b;
+		a; b;
+	}
+	function g(address view) public pure {}
+	function h(address pure) public pure {}
+	function i() public pure returns (address view) {}
+	function j() public pure returns (address pure) {}
+}
+// ----
+// TypeError: (14-26): Address types can only be payable or non-payable.
```

Comment:
This is more than one error per ``address *`` - ``ReferencesResolver`` seems to visit more than once here - I'll check why that is.